src/secp_ops.rs

```diff
@@ -46,31 +61,45 @@ pub fn op_secp256r1_verify(a: &mut Allocator, input: NodePtr, max_cost: Cost) ->
 // expects: pubkey msg sig
 pub fn op_secp256k1_verify(a: &mut Allocator, input: NodePtr, max_cost: Cost) -> Response {
     let cost = SECP256K1_VERIFY_COST;
-    check_cost(a, cost, max_cost)?;
+    check_cost(cost, max_cost)?;
 
     let [pubkey, msg, sig] = get_args::<3>(a, input, "secp256k1_verify")?;
 
     // first argument is sec1 encoded pubkey
     let pubkey = atom(a, pubkey, "secp256k1_verify pubkey")?;
-    let verifier = K1VerifyingKey::from_sec1_bytes(pubkey.as_ref())
-        .or_else(|_| err(input, "secp256k1_verify pubkey is not valid"))?;
+    let verifier = K1VerifyingKey::from_sec1_bytes(pubkey.as_ref()).map_err(|_| {
+        EvalErr::InvalidArg(
+            input,
+            "Secp256 Verify Error: pubkey is not valid".to_string(),
+        )
+    })?;
 
     // second arg is message
     let msg = atom(a, msg, "secp256k1_verify msg")?;
     if msg.as_ref().len() != 32 {
-        return err(input, "secp256k1_verify message digest is not 32 bytes");
+        Err(EvalErr::InvalidArg(
+            input,
+            "Secp256 Verify Error: message digest is not 32 bytes".to_string(),
+        ))?;
     }
 
     // third arg is a fixed-size signature
     let sig = atom(a, sig, "secp256k1_verify sig")?;
-    let sig = K1Signature::from_slice(sig.as_ref())
-        .or_else(|_| err(input, "secp256k1_verify sig is not valid"))?;
+    let sig = K1Signature::from_slice(sig.as_ref()).map_err(|_| {
+        EvalErr::InvalidArg(
+            input,
+            "Secp256 Verify Error: signature is not valid".to_string(),
+        )
+    })?;
 
     // verify signature
     let result = verifier.verify_prehash(msg.as_ref(), &sig);
 
     if result.is_err() {
-        err(input, "secp256k1_verify failed")
+        Err(EvalErr::InvalidArg(
```

Comment:
I think the validation error probably shouldn't be considered an invalid argument, but probably have its own error code. Same thing applies to Secp256r1 and BLS counterparts

Reply:
im going to put the 3 back in but let me know if you just want it as one error